tutorials/rag/evaluate-rag-flow.ipynb

```diff
@@ -842,6 +832,277 @@
+    "    \"\"\"Ask the user for clarifying information.\"\"\"\n",
+    "    # TODO: Implement this with your own logic for sending and receiving messages from the user\n",
+    "    print(\"AI: \", question)\n",
+    "    return input(\"User: \")\n",
+    "\n",
+    "def provide_answer(answer: str, reasoning: str, citation: str) -> str:\n",
+    "    \"\"\"Provide the answer to the user.\"\"\"\n",
+    "    # TODO: Implement this with your own logic for sending the final response to the user\n",
+    "    print(\n",
+    "        f\"Answer: {answer}\\nReasoning: {reasoning}\\nCitation: {citation}\"\n",
+    "    )\n"
+   ],
+   "id": "648d27e0c494a584"
+  },
+  {
+   "metadata": {
+    "jupyter": {
+     "is_executing": true
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "\n",
+    "import json\n",
+    "\n",
+    "model = \"gpt-4o-mini\"\n",
+    "temperature = 0\n",
+    "# define your Prompt template that contains variables for answer options and retrieved context\n",
+    "template = [\n",
+    "    {\n",
+    "        \"role\": \"system\",\n",
+    "        \"content\": \"\"\"Answer the question provided by the user factually. And reference the options for the answer:\n",
+    "\n",
+    "Options:\n",
+    "- {{option_A}}\n",
+    "- {{option_B}}\n",
+    "- {{option_C}}\n",
+    "- {{option_D}}\n",
+    "- {{option_E}}\n",
+    "\n",
+    "---\n",
+    "If the question is clearly stated, you can use the `retrieve_knowledge` tool to get more context before answering.\n",
+    "If you need more information, first call the `get_clarification` tool to ask the user for more details, before then calling the `retrieve_knowledge` tool.\n",
+    "\n",
+    "When you are happy with the context, call the `answer_question` tool to provide the answer to the user, where you should provide your reasoning and a clear citation.\n",
+    "\"\"\",\n",
+    "    }\n",
+    "]\n",
+    "\n",
+    "\n",
+    "def call_model(inputs: dict[str, str], messages: list[dict], trace_id: str) -> dict:\n",
+    "    \"\"\"Calls the model with the provided inputs and messages.\"\"\"\n",
+    "    \n",
+    "    # Populate the Prompt template\n",
+    "    start_time=datetime.now()\n",
+    "    populated_template_messages = humanloop.prompts.populate_template(template, inputs)\n",
+    "    \n",
+    "    # Call OpenAI to get response - note you can also instead manage your prompts on Humanloop and use our proxy:\n",
+    "    # https://humanloop.com/docs/v5/guides/prompts/call-prompt\n",
+    "    chat_completion= openai.chat.completions.create(\n",
+    "        model=model,\n",
+    "        temperature=temperature,\n",
+    "        messages=populated_template_messages + messages,\n",
+    "        # include the tool schemas so the model can make a call\n",
+    "        tools=tool_schemas,\n",
+    "    )\n",
+    "    # Log to Humanloop\n",
+    "    humanloop.prompts.log(\n",
+    "        path=\"evals_demo/medqa-answer\",\n",
```

Comment:
i'd updated the paths to our agreed sentence case.
so this should be "Evals demo/MedQA answer"

and the flow should should be "Evals demo/MedQA pipeline"

Reply:
No point continuing to review this given it has now been deleted